src/main.rs

```diff
@@ -121,36 +129,49 @@ fn try_main(args: Args) -> Result<()> {
                 }
             }
             Src::Paths(paths) => {
-                let mut files_changed = vec![];
-                let mut files_count = 0;
+                let (sender, receiver): (Sender<FormatResult>, Receiver<FormatResult>) = channel();
+
+                // Reducer, collect all the paths and statuses that have been seen
+                let reducer = thread::spawn(move || {
+                    let mut files_count = 0;
+                    let mut files_changed = 0;
+                    loop {
+                        match receiver.recv() {
```

Comment:
you can replace `loop` with `for (path, status) in receiver {`

Reply:
much nicer :)